emcc.py

```diff
@@ -1217,7 +1201,28 @@ def check(input_file):
           newargs.append('-disable-llvm-optzns')
 
       if not shared.Settings.LEGALIZE_JS_FFI:
-        assert shared.Building.is_wasm_only(), 'LEGALIZE_JS_FFI incompatible with RUNNING_JS_OPTS and non-wasm BINARYEN_METHOD.'
+        assert shared.Settings.WASM_ONLY, 'LEGALIZE_JS_FFI incompatible with RUNNING_JS_OPTS and non-wasm BINARYEN_METHOD.'
+
+      if shared.Settings.BINARYEN:
+        # determine wasm-only mode, now that all settings are settled on
+        shared.Settings.WASM_ONLY = shared.Building.is_wasm_only()
```

Comment:
Wait, so we add a `WASM_ONLY` flag that we assert gets set when we unset `LEGALIZE_JS_FFI`, but then we clobber it with our autodetected `shared.Building.is_wasm_only()` regardless?

We've promoted `WASM_ONLY` to a Setting so it's available to the js compiler, not because it's something users should be setting. Because of that I think we should move this to before the assert, and just always have it autodetected. Maybe add a note to settings.js, saying we ignore this if you try to set it.